File: songs/gnu-r/my-songs.r
library(tibble)

tibble(
  title = "Cantaloupe Island",
  tags  = list(list("60s", "jazz"))
) %>% add_row(
  title = "Let It Be",
  tags  = list(list("60s", "rock"))
) %>% add_row(
  title = "Knockin' on Heaven's Door",
  tags  = list(list("70s", "rock"))
) %>% add_row(
  title = "Emotion",
  tags  = list(list("70s", "pop"))
) %>% add_row(
  title = "The River"
) -> dataframe

File: songs/go/mysongs/mysongs.go
package mysongs

type Song struct {
	Title string
	Tags  []string
}

type Songs []Song

func GetSongs() Songs{
	return []Song{
		Song{"Cantaloupe Island",
			[]string{"60s", "jazz"}},
		Song{"Let It Be",
			[]string{"60s", "rock"}},
		Song{"Knockin' on Heaven's Door",
			[]string{"70s", "rock"}},
		Song{"Emotion",
			[]string{"70s", "pop"}},
		Song{"The River",
			nil},
	}
}

func (songs Songs) FlattenTags() []string {
	var tags []string

	for _, song := range songs {
		if song.Tags != nil {
			for _, tag := range song.Tags {
				tags = append(tags, tag)
			}
		}
	}

	return tags
}
